baseapp/abci.go

```diff
@@ -312,6 +314,40 @@ func (app *BaseApp) deliverTxWithoutEventHistory(req abci.RequestDeliverTx) (res
 	}
 }
 
+// checkHalt forces a state machine halt and attempts to kill the current
+// process if height or tm exceeds halt-height or halt-time respectively.
+func (app *BaseApp) checkHalt(height int64, tm time.Time) {
```

Comment:
Suggestions for further clarification:
```suggestion
// checkHalt forces a state machine halt and attempts to kill the current
// process if block height or timestamp exceeds halt-height or halt-time respectively.
func (app *BaseApp) checkHalt(blockHeight int64, blockTime time.Time) {
```